### PR TITLE
Add aten.histc decomposition (one_hot + sum)

### PR DESCRIPTION
**Pull request by kamalrajkannan78**
### Ticket

- https://github.com/tenstorrent/tt-xla/issues/5264

### Problem description

- `diffusiongemma-26B-A4B-it` crashes with `Fatal Python error: Segmentation fault`.
- `torch.histc` has no XLA/TTNN lowering, so torch_xla routes it to a CPU fallback. To move the op's inputs to CPU, the fallback force-materializes the in-flight (partial) XLA graph mid-stream, and compiling that partial graph segfaults inside torch_xla's `BuildStableHLOCompositePass` (in `_XLAC`). Full breakdown in the root-cause analysis of #5264.

### What's changed

- Added an `aten.histc` decomposition (`one_hot + sum`) in `python_package/tt_torch/backend/decompositions.py`, so `histc` lowers statically instead of falling back to CPU — this removes the mid-stream materialization and the segfault.
- Added `tests/torch/ops/test_histc.py` covering the model case plus histogram corner cases (fractional bins, out-of-range/NaN ignored, `value == max`, negative range, data-derived range).
- After this fix the model gets further and now fails with: `error: 'sdy.all_slice' op operates on axis "_axis_0" which is already bound by a parent sdy.manual_computation op`.

### Checklist
- [x] Verify the changes through local testing in n150

### Logs

- [jun17_diffgemma_before_fix.log](https://github.com/user-attachments/files/29102629/jun17_diffgemma_before_fix.log)
- [jun18_diffgemma_after_fix.log.zip](https://github.com/user-attachments/files/29102633/jun18_diffgemma_after_fix.log.zip)
- [jun22_histc_before_fix.log](https://github.com/user-attachments/files/29214089/jun22_histc_before_fix.log)
- [jun22_histc_after_fix.log](https://github.com/user-attachments/files/29214085/jun22_histc_after_fix.log)

### CODE REVIEW

File: python_package/tt_torch/backend/decompositions.py [hunk 1]
```diff
@@ -408,6 +408,19 @@ def _get_default_decomposition_ops() -> DecompositionOpsList:
     ]
 
 
+def histc(input, bins=100, min=0, max=0):
+    """Decomposition for aten.histc via one_hot + sum."""
+    # min == max == 0 means "use the tensor's own range" (torch.histc semantics).
+    lo, hi = (input.min(), input.max()) if min == 0 and max == 0 else (min, max)
```
Comment (vkovinicTT) [inline python_package/tt_torch/backend/decompositions.py]: Shouldn't we have more complete logic like [here](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cuda/SummaryOps.cu#L331) where we also cover cases where `min = max` and `> 0`?

Comment (kamalrajkannan78) [reply python_package/tt_torch/backend/decompositions.py]: Thanks, great catch! Addressed — switched `min == 0 and max == 0` to `min == max` so equal nonzero bounds (e.g. `min=max=5`) also use the tensor's own range, plus the constant-input widen (±1) from the kernel you linked. Added tests for both. updated the description with latest sanity log

For context: I opened this when `histc` was the blocker; only later, on current main while debugging the persistent segfault, I find it moves one op downstream to `torch._grouped_mm` ([details](https://github.com/tenstorrent/tt-xla/issues/5264#issuecomment-4769739755)), and that the cleaner DiffusionGemma fix is routing experts through the `tt_moe` backend (#5323). Still worth landing this one though — the `histc` decomposition is general and should help other models. Thanks for the review!

**Review by vkovinicTT — APPROVED** [review]

**Review by sdjukicTT — APPROVED** [review]
